src/components/include/utils/threads/thread.h

```diff
@@ -137,12 +132,12 @@ class Thread {
   /**
    * \brief Used to request actions from worker thread.
    */
-  ThreadCommand thread_command_;
+  volatile ThreadCommand thread_command_;
```

Comment:
@orhan-mehmedov volatile does not make it thread safe. Please investigate if you can use utils::Atomic